apis/graffiti.yaml

```
@@ -0,0 +1,122 @@
+get:
+  operationId: getGraffiti
+  summary: Get Graffiti
+  description: |
+    Get the graffiti for an individual validator. This graffiti is the one used by the
+    validator when proposing blocks. If no graffiti has been set explicitly for
+    a key then the process-wide default will be returned.
+  security:
+    - bearerAuth: []
+  tags:
+    - Graffiti
+  parameters:
+    - in: path
+      name: pubkey
+      schema:
+        $ref: "../keymanager-oapi.yaml#/components/schemas/Pubkey"
+      required: true
+  responses:
+    "200":
+      description: success response
+      content:
+        application/json:
+          schema:
+            title: ListGraffitiResponse
+            type: object
+            required: [data]
+            properties:
+              data:
+                type: object
+                required: [graffiti]
+                properties:
+                  pubkey:
+                    $ref: "../keymanager-oapi.yaml#/components/schemas/Pubkey"
+                  graffiti:
+                    $ref: "../keymanager-oapi.yaml#/components/schemas/Graffiti"
+    "400":
+      $ref: "../keymanager-oapi.yaml#/components/responses/BadRequest"
+    "401":
+      $ref: "../keymanager-oapi.yaml#/components/responses/Unauthorized"
+    "403":
+      $ref: "../keymanager-oapi.yaml#/components/responses/Forbidden"
+    "404":
+      $ref: "../keymanager-oapi.yaml#/components/responses/NotFound"
+    "500":
+      $ref: "../keymanager-oapi.yaml#/components/responses/InternalError"
+
+post:
+  operationId: setGraffiti
+  summary: Set Graffiti
+  description: |
+    Set the graffiti for an individual validator. This graffiti will be used instead
+    of the process-wide default by the validator when proposing blocks.
```

```suggestion
    Set the graffiti for an individual validator.
```
each api implementor may behave in different ways

---

> each api implementor may behave in different ways

I think it makes sense to outline more detailed behavior here. Ideally, we want all implementations to work the same way, i.e. validator specific graffiti has precedence over process-wide default. Otherwise, it becomes really unpredictable as a consumer of the API.

---

I'm not sure if all the clients have implemented this (the order of preference for loading the graffiti) the same way though, the exact behaviour isn't part of the spec AFAIK. 

I think as long as the response status code indicates the outcome (if it's a failure due to conflict), we should be ok?

---

> I think as long as the response status code indicates the outcome (if it's a failure due to conflict), we should be ok?

Based on how the API is supposed to work there can't be a conflict, the validator specific graffiti will be the one from the last update. In any case, if the update is successful the API returns a 202 status code.

---

> I think it makes sense to outline more detailed behavior here

Why? Not sure how that helps

---

> Why? Not sure how that helps

Besides what I already mentioned in https://github.com/ethereum/keymanager-APIs/pull/63#discussion_r1389568505 I don't have further arguments on why this might be helpful. 

I removed any comments regarding graffiti precedence for block proposal from API descriptions.